internal/storage/storage_handle_test.go

```diff
@@ -20,36 +39,79 @@ func getDefaultStorageClientConfig() (clientConfig storageClientConfig) {
 	}
 }
 
-func invokeAndVerifyStorageHandle(t *testing.T, sc storageClientConfig) {
+func TestStorageHandle(t *testing.T) { RunTests(t) }
+
+type StorageHandleTest struct {
+	fakeStorageServer *fakestorage.Server
+}
+
+var _ SetUpInterface = &StorageHandleTest{}
+var _ TearDownInterface = &StorageHandleTest{}
+
+func init() { RegisterTestSuite(&StorageHandleTest{}) }
+
+func (t *StorageHandleTest) SetUp(_ *TestInfo) {
+	var err error
+	t.fakeStorageServer, err = fakestorage.NewServerWithOptions(fakestorage.Options{
+		InitialObjects: []fakestorage.Object{
+			{
+				ObjectAttrs: fakestorage.ObjectAttrs{
+					BucketName: validBucketName,
+				},
+			},
+		},
+		Host: "127.0.0.1",
+		Port: 8081,
+	})
+	AssertEq(nil, err)
+}
+
+func (t *StorageHandleTest) TearDown() {
+	t.fakeStorageServer.Stop()
+}
+
+func (t *StorageHandleTest) invokeAndVerifyStorageHandle(sc storageClientConfig) {
 	handleCreated, err := NewStorageHandle(context.Background(), sc)
+	AssertEq(nil, err)
+	AssertNe(nil, handleCreated)
+}
 
-	if err != nil {
-		t.Errorf("Handle creation failure")
-	}
-	if handleCreated == nil {
-		t.Fatalf("Storage handle is null")
-	}
-	if nil == handleCreated.client {
-		t.Fatalf("Storage client handle is null")
-	}
+func (t *StorageHandleTest) TestBucketHandleWhenBucketExists() {
+	fakeClient := t.fakeStorageServer.Client()
+	fakeStorageClient := &storageClient{client: fakeClient}
```

Comment:
nit: dont call it fakeStorageClient, rename it to storageClient.
It gives the impression that we are creating a fake instance and invoking test method on it. but we are creating a storageClient instance which has fakeClient.